### PR TITLE
Add cql-json support for ES

### PR DESCRIPTION
**Pull request by francbartoli**
This pull request replces #670 and adds an early implementation of CQL support for the Elasticsearch provider with the CQL-JSON dialect. It supports the following type of queries:

- [X] `between` predicate query
- [X] Logical `and` query with `between` and `eq` expression
- [x] Spatial query with `bbox`

### CODE REVIEW

File: pygeoapi/api.py [hunk 1]
```diff
@@ -1678,7 +1681,9 @@ def post_collection_items(
         LOGGER.debug('Processing headers')
 
         LOGGER.debug('Processing request content-type header')
-        if request_headers.get('Content-Type') != 'application/query-cql-json':
+        if (request_headers.get(
+            'Content-Type') or request_headers.get(
+                'content-type')) != 'application/query-cql-json':
```
[the remark this answers is not in the document]
Comment (francbartoli) [reply pygeoapi/api.py:1686]: I would postpone this to a different pull request

**Review by tomkralidis — COMMENTED** [review]
Looks great @francbartoli.  My only questions are around use of nest_asyncio (does this have to be in `pygeoapi/api.py` or can we use in the starlette specific code?) and future-annotations.

**Comment by francbartoli** [comment]
> Looks great @francbartoli. My only questions are around use of nest_asyncio (does this have to be in `pygeoapi/api.py` or can we use in the starlette specific code?) and future-annotations.

@tomkralidis nest_asyncio fixes the `APIRequest` class with starlette (which was broken before) and future-annotations can be abandoned once we'll deprecate python 3.6

**Comment by tomkralidis** [comment]
Excellent work @francbartoli !

**Pull request merged**